File: @zlliu246/playlists/can-you-solve-this-in-one-line/c-plus-plus-programming-language/src/triangle3.cpp
/*
* std::cin
* std::cout
* std::endl
*/
#include <iostream>

// EXIT_SUCCESS
#include <cstdlib>

// std::format
#include <format>

/**
 * Function declaration
*/
void
triangle3(int rows);

/*
* C++ program execution starts
* from main() function
*/
int
main()
{
    std::cout<<
    std::format(
      "Enter number of rows: "
    )<<std::endl;
    int rows {};
    std::cin>>rows;
    /*
    * Function call
    */
    triangle3(rows);
  /*
  * Return EXIT_SUCCESS to the
  * underlying operating-system
  */
  return EXIT_SUCCESS;
}

/**
 * Function definition
*/
void
triangle3(int rows)
{
    for (int row {rows} ; row >= 1 ; row -= 1)
    {
      for (int column {rows} ; column > rows - row ; column -= 1)
      {
        std::cout<<
        std::format(
          "{}",
          column
        );
      }
      std::cout<<std::endl;
    }
  return;
}

File: @zlliu246/playlists/can-you-solve-this-in-one-line/rust-programming-language/triangle3/src/triangle3.rs
use std::io::stdin;

fn main() -> () {
    println!(
      "Enter number of rows: "
    );
    let mut string_buffer : String = String::new();
    stdin().read_line(&mut string_buffer).expect("Error occured when reading from standard-input");
    let rows : u32 = string_buffer.trim().parse().expect("Error occurred when converting a string-literal to an integer-literal");
    //
    // Function call
    //
    triangle3(rows);
  //
  // Return an empty tuple also
  // known as aan unit to the
  // underlying operating-system
  //
  return;
}

//
// Function definition
//
fn triangle3( rows : u32 ) -> () {
    let mut row = rows;
    while row >= 1 {
      let mut column = rows;
      while column > rows - row {
          print!("{}",column);
        column -= 1;
      }
      row -= 1;
      println!();  
    }
  //
  // Return an empty tuple also
  // known as an unit to the
  // caller
  //
  return;  
}